content/id/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins.md

```diff
@@ -0,0 +1,159 @@
+    }
+  ]
+}
+```
+
+#### Dukungan pembentukan lalu-lintas
+
+_Plugin_ jaringan CNI juga mendukung masuknya Pod dan membentuk lalu-lintas keluar. Kamu dapat menggunakan _plugin_ resmi [_bandwidth_](https://github.com/containernetworking/plugins/tree/master/plugins/meta/bandwidth) yang ditawarkan oleh tim _plugin_ CNI atau menggunakan _plugin_ kamu sendiri dengan fungsi kontrol _bandwidth_.
+
+Jika kamu ingin mengaktifkan pembentukan lalu-lintas, kamu harus menambahkan _plugin_ `bandwidth` ke berkas konfigurasi CNI kamu (bawaan `/etc/cni/ net.d`).
+
+```json
+{
+  "name": "k8s-pod-network",
+  "cniVersion": "0.3.0",
+  "plugins": [
+    {
+      "type": "calico",
+      "log_level": "info",
+      "datastore_type": "kubernetes",
+      "nodename": "127.0.0.1",
+      "ipam": {
+        "type": "host-local",
+        "subnet": "usePodCidr"
+      },
+      "policy": {
+        "type": "k8s"
+      },
+      "kubernetes": {
+        "kubeconfig": "/etc/cni/net.d/calico-kubeconfig"
+      }
+    },
+    {
+      "type": "bandwidth",
+      "capabilities": {"bandwidth": true}
+    }
+  ]
+}
+```
+
+Sekarang kamu dapat menambahkan anotasi `kubernetes.io/ingress-bandwidth` dan `kubernetes.io/egress-bandwidth` ke Pod kamu.
+Contoh:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    kubernetes.io/ingress-bandwidth: 1M
+    kubernetes.io/egress-bandwidth: 1M
+...
+```
+
+### Kubenet
+
+Kubenet adalah _plugin_ jaringan yang sangat mendasar dan sederhana, hanya untuk Linux. Ini sendiri, tidak dengan sendirinya, mengimplementasikan fitur-fitur yang lebih canggih seperti jaringan _cross-node_ atau kebijakan jaringan. Ini biasanya digunakan bersamaan dengan penyedia _cloud_ yang menetapkan aturan _routing_ untuk komunikasi antar Node, atau dalam lingkungan Node tunggal.
+
+Kubenet membuat _bridge_ Linux bernama `cbr0` dan membuat pasangan _veth_ untuk setiap Pod dengan ujung _host_ dari setiap pasangan yang terhubung ke `cbr0`. Ujung Pod dari pasangan diberi alamat IP yang dialokasikan dari rentang yang ditetapkan untuk Node baik melalui konfigurasi atau oleh controller-manager. `cbr0` memiliki MTU yang cocok dengan MTU terkecil dari antarmuka normal yang diaktifkan pada _host_.
+
+_Plugin_ memerlukan beberapa hal:
+
+* Diperlukan _plugin_ CNI `bridge`, `lo` dan `host-local` standar, pada versi minimum 0.2.0. Kubenet pertama-tama akan mencari mereka di `/opt/cni/bin`. Tentukan `cni-bin-dir` untuk menyediakan lokasi pencarian tambahan. Hasi pencarian pertama yang akan digunakan.
+* Kubelet harus dijalankan dengan argumen `--network-plugin=kubenet` untuk mengaktifkan _plugin_
+* Kubelet juga harus dijalankan dengan argumen `--non-masquerade-cidr=<clusterCidr>` untuk memastikan lalu-lintas ke IP di luar rentang ini akan menggunakan IP _masquerade_.
+* Node harus diberi subnet IP melalui perintah kubelet `--pod-cidr` atau perintah _controller-manager_ `--allocate-node-cidrs=true --cluster-cidr=<cidr>`.
```

Comment:
```suggestion
* Node harus diberi subnet IP melalui perintah kubelet `--pod-cidr` atau perintah controller-manager `--allocate-node-cidrs=true --cluster-cidr=<cidr>`.
```

Controller-manager is common a Kubernetes sytem component, no need to italicize.

Reply:
okay noted. done